cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaBridgeClusterTest.java

```diff
@@ -182,31 +190,44 @@ public void testGenerateService()   {
-        assertThat(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getPorts().get(0).getContainerPort(), is(Integer.valueOf(KafkaBridgeCluster.DEFAULT_REST_API_PORT)));
-        assertThat(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getPorts().get(0).getName(), is(KafkaBridgeCluster.REST_API_PORT_NAME));
-        assertThat(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getPorts().get(0).getProtocol(), is("TCP"));
-        assertThat(dep.getSpec().getStrategy().getType(), is("RollingUpdate"));
-        assertThat(dep.getSpec().getStrategy().getRollingUpdate().getMaxSurge().getIntVal(), is(Integer.valueOf(1)));
-        assertThat(dep.getSpec().getStrategy().getRollingUpdate().getMaxUnavailable().getIntVal(), is(Integer.valueOf(0)));
-        assertThat(AbstractModel.containerEnvVars(dep.getSpec().getTemplate().getSpec().getContainers().get(0)).get(KafkaBridgeCluster.ENV_VAR_KAFKA_BRIDGE_TLS), is(nullValue()));
+
+
+
+
+        ObjectMeta metadata = dep.getMetadata();
+        assertThat(metadata.getName(), is(KafkaBridgeResources.deploymentName(cluster)));
+        assertThat(metadata.getNamespace(), is(namespace));
+        assertThat(metadata.getLabels(), is(expectedDeploymentLabels));
+
+        DeploymentSpec spec = dep.getSpec();
+        assertThat(spec.getSelector().getMatchLabels(), is(expectedSelectorLabels()));
+        assertThat(spec.getReplicas(), is(replicas));
+        assertThat(spec.getTemplate().getMetadata().getLabels(), is(expectedDeploymentLabels));
+        assertThat(spec.getStrategy().getType(), is("RollingUpdate"));
+        assertThat(spec.getStrategy().getRollingUpdate().getMaxSurge().getIntVal(), is(1));
+        assertThat(spec.getStrategy().getRollingUpdate().getMaxUnavailable().getIntVal(), is(0));
+
+        assertThat(spec.getTemplate().getSpec().getContainers().size(), is(1));
+
+        Container container = spec.getTemplate().getSpec().getContainers().get(0);
+        assertThat(container.getName(), is(KafkaBridgeResources.deploymentName(cluster)));
+        assertThat(container.getImage(), is(kbc.image));
+        assertThat(container.getEnv(), is(getExpectedEnvVars()));
+        assertThat(container.getLivenessProbe().getInitialDelaySeconds(), is(healthDelay));
+        assertThat(container.getLivenessProbe().getTimeoutSeconds(), is(healthTimeout));
+        assertThat(container.getReadinessProbe().getInitialDelaySeconds(), is(healthDelay));
+        assertThat(container.getReadinessProbe().getTimeoutSeconds(), is(healthTimeout));
+
+        assertThat(container.getPorts().size(), is(1));
+
+        ContainerPort containerPort = container.getPorts().get(0);
+        assertThat(containerPort.getContainerPort(), is(KafkaBridgeCluster.DEFAULT_REST_API_PORT));
+        assertThat(containerPort.getName(), is(KafkaBridgeCluster.REST_API_PORT_NAME));
+        assertThat(containerPort.getProtocol(), is("TCP"));
+
```

Comment:
Hi @scholzj, I wanted to check that this refactoring like many others in this file), pulling out commonly reused expressions that could be expressed as a single variable, doesn't go too far in your opinion. 

I've held back on further similar refactorings after this file, as I don't want to race ahead without your blessing.

Let me know :)

Reply:
I'm probably fine with this. Although I do not think this was necessarily needed. 

I think there are several aspects to handle:
* Code readability
* Performance / resource consumption (The UT/IT tests now run for 30-40 minutes, so saving time is nice)

My particular concern in this respect are mainly the asserts where we do `stream()` several times in a row and search for exactly the same thing just to assert several different parts. That seems rather wasteful. I guess the getters being called here again and again should be less wasteful (although to be honest I'm just guessing - I'm not really expert on this kind of stuff).

Reply:
Ah yep, good point on the reusage of stream().